### ISSUE TITLE
[a11y] VoiceOver focus doesn't work well when there's a semantic overlap 

### ISSUE THREAD

**Issue by romgrm** [flutter/flutter#131780]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Hi ! 

In the video showing the bug, I have 2 RawApplePayButtons which are supposed to open a bottomSheet when onPressed is triggered.

The first Apple Pay button is used "conventionally", i.e. there are no other Semantics wrapped around it, so the widget works fine.

The second button is wrapped by a SizedBox(), itself wrapped by a Semantics(). When the RawApplePayButton is tapped, the callback is not triggered.
However, as you can see in the video, if I focus on my SizedBox() and click on it, the onPressed on my RawApplePayButton is called, which shouldn't be the case.

In my Apple Pay button KO, the focus seems to be directly on the PKPaymentButton and so Flutter's onPressed callback isn't triggered.

The bug seems to be only from iOS 16.5 onwards, and only when voice is used. (I tested on iPhone 14 pro on 16.5.1)

I've already opened an issue on the google-pay github but they told me that their library used the standard apple pay button behavior. 

If you want to reproduce you can take the code sample and install the [pay package](https://pub.dev/packages/pay)

### Expected results

onPressed on the ApplePay button is triggered when it is tapped while the VoiceOver focus is on it. 

### Actual results

onPressed on the ApplePay button is triggered while VoiceOver focus is on the parent (Container).

### Code sample

<details><summary>Code sample</summary>

```dart
class ApplePayIssue extends StatefulWidget {
  const ApplePayIssue({super.key});

  @override
  State<ApplePayIssue> createState() => _ApplePayIssueState();
}

class _ApplePayIssueState extends State<ApplePayIssue> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Text("apple pay button OK"),
          RawApplePayButton(
            style: ApplePayButtonStyle.black,
            type: ApplePayButtonType.plain,
            onPressed: () {
              showModalBottomSheet(
                  context: context,
                  builder: (context) {
                    return const SizedBox(
                      height: 200,
                      child: Text("apple pay modal OK"),
                    );
                  });
            },
          ),
          const SizedBox(
            height: 50,
          ),
          const Text("apple pay button KO"),
          Semantics(
            label: "paiement card",
            excludeSemantics: false,
            explicitChildNodes: true,
            child: Container(
              color: Colors.red,
              padding: const EdgeInsets.all(10),
              child: RawApplePayButton(
                style: ApplePayButtonStyle.black,
                type: ApplePayButtonType.plain,
                onPressed: () {
                  showModalBottomSheet(
                      context: context,
                      builder: (context) {
                        return const SizedBox(
                          height: 200,
                          child: Text("apple pay modal OK"),
                        );
                      });
                },
              ),
            ),
          ),
        ],
      ),
    ));
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/72022928/30d5a150-189c-4303-b2de-fa743b26ae65

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[!] Flutter (Channel stable, 3.7.12, on macOS 13.4 22F66 darwin-arm64, locale fr-FR)
    • Flutter version 3.7.12 on channel stable at /Users/romain_greaume/fvm/versions/3.7.12
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.0.7/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/romain_greaume/fvm/versions/3.7.12. Consider adding /Users/romain_greaume/fvm/versions/3.7.12/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (4 months ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/romain_greaume/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 73.1.1
    • Dart plugin version 231.9065

[✓] VS Code (version 1.80.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 115.0.5790.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available
    ```

</details>

**Comment by romgrm** [comment]
Hi @huycozy  :) 

Thanks for the answer. 
So I upgraded flutter to be in 3.13.0-19.0.pre.78, the issue is always there when you use the ApplePay button on the lib. 
However, if I use an ElevatedButton for example, the issue is no longer there. (you can see the version upgrade through the modals/buttons which have changed their appearance)

https://github.com/flutter/flutter/assets/72022928/fc28a1d1-6e48-4e8d-966c-2c1f036421b0

https://github.com/flutter/flutter/assets/72022928/86489519-3322-4aeb-a6c4-f654e49c4123

however, strangely, in 3.7.12, with the ElevatedButton I also have a KO. If I tap on the red card it opens the modal, as well as when I click on the button. The code hasn't changed 


https://github.com/flutter/flutter/assets/72022928/ca5a7573-9554-4ec1-b85c-7fa9487d30d5

When I'm in KO in my logs I get this message. I've looked a bit and it seems to be a problem on the iOS side so that voiceOver can find each nested element. 

```
[Accessibility] Went all the way up the container chain from @, but could not find any container that was one of the ordered children of @. This may be acceptable if it happened right around a layout change, but it would be best to double check by swiping left/right to see if you can get to all elements.
```

I have to admit that I'm a bit stuck here, I've the impression that the issue is really between flutter and iOS in terms of managing nested Semantics. 

What do you think?

**Comment by huycozy** [comment]
I took a look at the `RawApplePayButton` code and see it's using a platform view with `UiKitView`. I have created a minimal code sample here [a11y-ios-131780](https://github.com/huycozy/platformviews-demo/tree/a11y-ios-131780). Unfortunately, I don't have a higher iOS version to test (the issue doesn't reproduce with my iOS 15.7). Can you pull this sample code and check it on your device? If it's reproducible, it could be an OS-specific + platform-views + a11y issue.

**Comment by romgrm** [comment]
@huycozy 
I've pulled the repo but I've got an error on FLNativeViewFactory, the linter tells me it can't find it in the scope. 
I do have the UIKit and Flutter imports. 
I saw that you had gitIgnore your podfile, I've created one but I don't know if a lib is missing? 
It seems to me that FLNativeViewFactory is directly integrated into UIKit so I don't understand what the problem is... 

Did you take over the code from  https://docs.flutter.dev/platform-integration/ios/platform-views ?

**Comment by huycozy** [comment]
>Did you take over the code from https://docs.flutter.dev/platform-integration/ios/platform-views ?

Yes, basically that's the code base. I have just pushed the remaining local changes to that branch, please pull and check if it works. 

Also, you can check the [commit](https://github.com/huycozy/platformviews-demo/commit/cf7a3191ca9a7fdc1fec2d8df62ed57745bde6a0) which updates to that code base and replace the text with a button to reproduce this issue.

**Comment by Jerem42** [comment]
Hey @huycozy, from my tests, the issue is linked to the native Apple Pay Button (PKPaymentButton), something might have changed with iOS 16.5 and the communication between Flutter and the button might have changed. I tried to change it to a normal UIButton in the pay plugin locally and it's working again.

https://github.com/google-pay/flutter-plugin/blob/af2e70d121778337bf4bda9b5ef12ffbe4df664d/pay_ios/ios/Classes/ApplePayButtonView.swift#L141
![image](https://github.com/flutter/flutter/assets/15171406/5dec330b-dc2e-43c1-afcc-93b445edfca0)

**Comment by huycozy** [comment]
@Jerem42 I haven't dug into it so I'm not sure how `PKPaymentButton` is built, I guess it's a native UI component like UIButton but more complex. I think if it can be solved from `google-pay` package side, it should be reported there for better support (I see there is an existing issue but closed, you can forward this info to that issue).
